### PR TITLE
Document Appearances.

### PR DESCRIPTION
**Pull request by DavidMansolino**
Since we start to have a nice set of appearances (#165 will add a few more) it is important to document them:
  - [x] Extend icon studio to generate appearance doc image + icons.
  - [x] Extend objects doc generator to generate appearance doc too.
  - [x] Fix texturetransform and color issue.

https://www.cyberbotics.com/doc/guide/appearances?version=document-appearances

### CODE REVIEW

**Comment by DavidMansolino** [comment]
test me

**Comment by DavidMansolino** [comment]
:+1:

File: docs/guide/appearances.md [hunk 1]
```diff
@@ -0,0 +1,714 @@
...
+
+- `textureTransform`: Defines an optional 2d texture transform.
+
+- `environmentMap`: Defines an optional `Cubemap` node that can override the world's skybox for this object.
+
+- `IBLStrength`: Defines the strength of ambient lighting from the Cubemap node.
+
+## GlossyCarPaint
+
+A glossy car paint material. The color can be selected `baseColor` field. Useful with any of the vehicle PROTOs.
```
Comment (omichel) [inline docs/guide/appearances.md]:
```suggestion
A glossy car paint material. The color can be selected in the `baseColor` field. Useful with any of the vehicle PROTOs.
```

File: docs/guide/appearances.md [hunk 2]
```diff
@@ -0,0 +1,714 @@
...
+
+- `textureTransform`: Defines an optional 2d texture transform.
+
+- `environmentMap`: Defines an optional `Cubemap` node that can override the world's skybox for this object.
+
+- `IBLStrength`: Defines the strength of ambient lighting from the Cubemap node.
+
+## MatteCarPaint
+
+A matte car paint material. The color can be selected `baseColor` field. Useful with any of the vehicle PROTOs.
```
Comment (omichel) [inline docs/guide/appearances.md]:
```suggestion
A matte car paint material. The color can be selected in the `baseColor` field. Useful with any of the vehicle PROTOs.
```

File: docs/guide/appearances.md [hunk 3]
```diff
@@ -0,0 +1,714 @@
...
+> **File location**: "WEBOTS\_HOME/projects/appearances/protos/MetalPipePaint.proto"
+
+> **License**: Copyright Cyberbotics Ltd. Licensed for use only with Webots.
+[More information.](https://cyberbotics.com/webots_assets_license)
+
+### MetalPipePaint Field Summary
+
+- `textureTransform`: Defines an optional 2d texture transform.
+
+- `environmentMap`: Defines an optional `Cubemap` node that can override the world's skybox for this object.
```
Comment (omichel) [inline docs/guide/appearances.md]:
```suggestion
- `environmentMap`: Defines an optional `Cubemap` node overriding the world's skybox for this object.
```

File: docs/guide/appearances.md [hunk 4]
```diff
@@ -0,0 +1,714 @@
...
+> **License**: Copyright Cyberbotics Ltd. Licensed for use only with Webots.
+[More information.](https://cyberbotics.com/webots_assets_license)
+
+### Asphalt Field Summary
+
+- `colorOverride`: Defines the default color multiplied with the texture color.
+
+- `textureTransform`: Defines an optional 2d texture transform.
+
+- `environmentMap`: Defines an optional `Cubemap` node that can override the world's skybox for this object.
```
Comment (omichel) [inline docs/guide/appearances.md]:
```suggestion
- `environmentMap`: Defines an optional `Cubemap` node overriding the world's skybox for this object.
```

File: docs/guide/appearances.md [hunk 5]
```diff
@@ -0,0 +1,714 @@
...
+> **License**: Copyright Cyberbotics Ltd. Licensed for use only with Webots.
+[More information.](https://cyberbotics.com/webots_assets_license)
+
+### BrushedAluminium Field Summary
+
+- `colorOverride`: Defines the default color multiplied with the texture color.
+
+- `textureTransform`: Defines an optional 2d texture transform.
+
+- `environmentMap`: Defines an optional `Cubemap` node that can override the world's skybox for this object.
```
Comment (omichel) [inline docs/guide/appearances.md]:
```suggestion
- `environmentMap`: Defines an optional `Cubemap` node overriding the world's skybox for this object.
```

File: docs/guide/appearances.md [hunk 6]
```diff
@@ -0,0 +1,714 @@
...
+> **License**: Copyright Cyberbotics Ltd. Licensed for use only with Webots.
+[More information.](https://cyberbotics.com/webots_assets_license)
+
+### CarpetFibers Field Summary
+
+- `colorOverride`: Defines the default color multiplied with the texture color.
+
+- `textureTransform`: Defines an optional 2d texture transform.
+
+- `environmentMap`: Defines an optional `Cubemap` node that can override the world's skybox for this object.
```
Comment (omichel) [inline docs/guide/appearances.md]:
```suggestion
- `environmentMap`: Defines an optional `Cubemap` node overriding the world's skybox for this object.
```

File: docs/guide/appearances.md [hunk 7]
```diff
@@ -0,0 +1,714 @@
...
+> **File location**: "WEBOTS\_HOME/projects/appearances/protos/ChequeredParquetry.proto"
+
+> **License**: Copyright Cyberbotics Ltd. Licensed for use only with Webots.
+[More information.](https://cyberbotics.com/webots_assets_license)
+
+### ChequeredParquetry Field Summary
+
+- `textureTransform`: Defines an optional 2d texture transform.
+
+- `environmentMap`: Defines an optional `Cubemap` node that can override the world's skybox for this object.
```
Comment (omichel) [inline docs/guide/appearances.md]:
```suggestion
- `environmentMap`: Defines an optional `Cubemap` node overriding the world's skybox for this object.
```

File: docs/guide/appearances.md [hunk 8]
```diff
@@ -0,0 +1,714 @@
...
+> **License**: Copyright Cyberbotics Ltd. Licensed for use only with Webots.
+[More information.](https://cyberbotics.com/webots_assets_license)
+
+### GlossyCarPaint Field Summary
+
+- `baseColor`: Defines the paint's color.
+
+- `textureTransform`: Defines an optional 2d texture transform.
+
+- `environmentMap`: Defines an optional `Cubemap` node that can override the world's skybox for this object.
```
Comment (omichel) [inline docs/guide/appearances.md]:
```suggestion
- `environmentMap`: Defines an optional `Cubemap` node overriding the world's skybox for this object.
```

File: docs/guide/appearances.md [hunk 9]
```diff
@@ -0,0 +1,714 @@
...
+> **License**: Copyright Cyberbotics Ltd. Licensed for use only with Webots.
+[More information.](https://cyberbotics.com/webots_assets_license)
+
+### MatteCarPaint Field Summary
+
+- `baseColor`: Defines the paint's color.
+
+- `textureTransform`: Defines an optional 2d texture transform.
+
+- `environmentMap`: Defines an optional `Cubemap` node that can override the world's skybox for this object.
```
Comment (omichel) [inline docs/guide/appearances.md]:
```suggestion
- `environmentMap`: Defines an optional `Cubemap` node overriding the world's skybox for this object.
```

File: docs/guide/appearances.md [hunk 10]
```diff
@@ -0,0 +1,714 @@
...
+> **License**: Copyright Cyberbotics Ltd. Licensed for use only with Webots.
+[More information.](https://cyberbotics.com/webots_assets_license)
+
+### OldSteel Field Summary
+
+- `colorOverride`: Defines the default color multiplied with the texture color.
+
+- `textureTransform`: Defines an optional 2d texture transform.
+
+- `environmentMap`: Defines an optional `Cubemap` node that can override the world's skybox for this object.
```
Comment (omichel) [inline docs/guide/appearances.md]:
```suggestion
- `environmentMap`: Defines an optional `Cubemap` node overriding the world's skybox for this object.
```

File: docs/guide/appearances.md [hunk 11]
```diff
@@ -0,0 +1,714 @@
...
+> **License**: Copyright Cyberbotics Ltd. Licensed for use only with Webots.
+[More information.](https://cyberbotics.com/webots_assets_license)
+
+### PaintedWood Field Summary
+
+- `colorOverride`: Defines the default color multiplied with the texture color.
+
+- `textureTransform`: Defines an optional 2d texture transform.
+
+- `environmentMap`: Defines an optional `Cubemap` node that can override the world's skybox for this object.
```
Comment (omichel) [inline docs/guide/appearances.md]:
```suggestion
- `environmentMap`: Defines an optional `Cubemap` node overriding the world's skybox for this object.
```

File: docs/guide/appearances.md [hunk 12]
```diff
@@ -0,0 +1,714 @@
...
+> **License**: Copyright Cyberbotics Ltd. Licensed for use only with Webots.
+[More information.](https://cyberbotics.com/webots_assets_license)
+
+### Parquetry Field Summary
+
+- `colorOverride`: Defines the default color multiplied with the texture color.
+
+- `textureTransform`: Defines an optional 2d texture transform.
+
+- `environmentMap`: Defines an optional `Cubemap` node that can override the world's skybox for this object.
```
Comment (omichel) [inline docs/guide/appearances.md]:
```suggestion
- `environmentMap`: Defines an optional `Cubemap` node overriding the world's skybox for this object.
```

File: docs/guide/appearances.md [hunk 13]
```diff
@@ -0,0 +1,714 @@
...
+> **File location**: "WEBOTS\_HOME/projects/appearances/protos/RedBricks.proto"
+
+> **License**: Copyright Cyberbotics Ltd. Licensed for use only with Webots.
+[More information.](https://cyberbotics.com/webots_assets_license)
+
+### RedBricks Field Summary
+
+- `textureTransform`: Defines an optional 2d texture transform.
+
+- `environmentMap`: Defines an optional `Cubemap` node that can override the world's skybox for this object.
```
Comment (omichel) [inline docs/guide/appearances.md]:
```suggestion
- `environmentMap`: Defines an optional `Cubemap` node overriding the world's skybox for this object.
```

File: docs/guide/appearances.md [hunk 14]
```diff
@@ -0,0 +1,714 @@
...
+> **License**: Copyright Cyberbotics Ltd. Licensed for use only with Webots.
+[More information.](https://cyberbotics.com/webots_assets_license)
+
+### RoughConcrete Field Summary
+
+- `colorOverride`: Defines the default color multiplied with the texture color.
+
+- `textureTransform`: Defines an optional 2d texture transform.
+
+- `environmentMap`: Defines an optional `Cubemap` node that can override the world's skybox for this object.
```
Comment (omichel) [inline docs/guide/appearances.md]:
```suggestion
- `environmentMap`: Defines an optional `Cubemap` node overriding the world's skybox for this object.
```

File: docs/guide/appearances.md [hunk 15]
```diff
@@ -0,0 +1,714 @@
...
+> **License**: Copyright Cyberbotics Ltd. Licensed for use only with Webots.
+[More information.](https://cyberbotics.com/webots_assets_license)
+
+### RoughOak Field Summary
+
+- `colorOverride`: Defines the default color multiplied with the texture color.
+
+- `textureTransform`: Defines an optional 2d texture transform.
+
+- `environmentMap`: Defines an optional `Cubemap` node that can override the world's skybox for this object.
```
Comment (omichel) [inline docs/guide/appearances.md]:
```suggestion
- `environmentMap`: Defines an optional `Cubemap` node overriding the world's skybox for this object.
```

File: docs/guide/appearances.md [hunk 16]
```diff
@@ -0,0 +1,714 @@
...
+> **License**: Copyright Cyberbotics Ltd. Licensed for use only with Webots.
+[More information.](https://cyberbotics.com/webots_assets_license)
+
+### RoughPine Field Summary
+
+- `colorOverride`: Defines the default color multiplied with the texture color.
+
+- `textureTransform`: Defines an optional 2d texture transform.
+
+- `environmentMap`: Defines an optional `Cubemap` node that can override the world's skybox for this object.
```
Comment (omichel) [inline docs/guide/appearances.md]:
```suggestion
- `environmentMap`: Defines an optional `Cubemap` node overriding the world's skybox for this object.
```

File: docs/guide/appearances.md [hunk 17]
```diff
@@ -0,0 +1,714 @@
...
+> **License**: Copyright Cyberbotics Ltd. Licensed for use only with Webots.
+[More information.](https://cyberbotics.com/webots_assets_license)
+
+### Roughcast Field Summary
+
+- `colorOverride`: Defines the default color multiplied with the texture color.
+
+- `textureTransform`: Defines an optional 2d texture transform.
+
+- `environmentMap`: Defines an optional `Cubemap` node that can override the world's skybox for this object.
```
Comment (omichel) [inline docs/guide/appearances.md]:
```suggestion
- `environmentMap`: Defines an optional `Cubemap` node overriding the world's skybox for this object.
```

File: docs/guide/appearances.md [hunk 18]
```diff
@@ -0,0 +1,714 @@
...
+> **License**: Copyright Cyberbotics Ltd. Licensed for use only with Webots.
+[More information.](https://cyberbotics.com/webots_assets_license)
+
+### SandyGround Field Summary
+
+- `colorOverride`: Defines the default color multiplied with the texture color.
+
+- `textureTransform`: Defines an optional 2d texture transform.
+
+- `environmentMap`: Defines a `Cubemap` node that can override the world's skybox for this object.
```
Comment (omichel) [inline docs/guide/appearances.md]:
```suggestion
- `environmentMap`: Defines a `Cubemap` node overriding the world's skybox for this object.
```

File: docs/guide/appearances.md [hunk 19]
```diff
@@ -0,0 +1,714 @@
...
+> **License**: Copyright Cyberbotics Ltd. Licensed for use only with Webots.
+[More information.](https://cyberbotics.com/webots_assets_license)
+
+### ShinyLeather Field Summary
+
+- `colorOverride`: Defines the leather's color.
+
+- `textureTransform`: Defines an optional 2d texture transform.
+
+- `environmentMap`: Defines a `Cubemap` node that can override the world's skybox for this object.
```
Comment (omichel) [inline docs/guide/appearances.md]:
```suggestion
- `environmentMap`: Defines a `Cubemap` node overriding the world's skybox for this object.
```

File: docs/guide/appearances.md [hunk 20]
```diff
@@ -0,0 +1,714 @@
...
+> **License**: Copyright Cyberbotics Ltd. Licensed for use only with Webots.
+[More information.](https://cyberbotics.com/webots_assets_license)
+
+### StonePavement Field Summary
+
+- `colorOverride`: Defines the default color multiplied with the texture color.
+
+- `textureTransform`: Defines an optional 2d texture transform.
+
+- `environmentMap`: Defines an optional `Cubemap` node that can override the world's skybox for this object.
```
Comment (omichel) [inline docs/guide/appearances.md]:
```suggestion
- `environmentMap`: Defines an optional `Cubemap` node overriding the world's skybox for this object.
```

File: docs/guide/appearances.md [hunk 21]
```diff
@@ -0,0 +1,714 @@
...
+> **License**: Copyright Cyberbotics Ltd. Licensed for use only with Webots.
+[More information.](https://cyberbotics.com/webots_assets_license)
+
+### ThreadMetalPlate Field Summary
+
+- `colorOverride`: Defines the default color multiplied with the texture color.
+
+- `textureTransform`: Defines an optional 2d texture transform.
+
+- `environmentMap`: Defines an optional `Cubemap` node that can override the world's skybox for this object.
```
Comment (omichel) [inline docs/guide/appearances.md]:
```suggestion
- `environmentMap`: Defines an optional `Cubemap` node overriding the world's skybox for this object.
```

File: docs/guide/appearances.md [hunk 22]
```diff
@@ -0,0 +1,714 @@
...
+> **License**: Copyright Cyberbotics Ltd. Licensed for use only with Webots.
+[More information.](https://cyberbotics.com/webots_assets_license)
+
+### VarnishedPine Field Summary
+
+- `colorOverride`: Defines the default color multiplied with the texture color.
+
+- `textureTransform`: Defines an optional 2d texture transform.
+
+- `environmentMap`: Defines an optional `Cubemap` node that can override the world's skybox for this object.
```
Comment (omichel) [inline docs/guide/appearances.md]:
```suggestion
- `environmentMap`: Defines an optional `Cubemap` node overriding the world's skybox for this object.
```

**Comment by DavidMansolino** [comment]
test me

**Comment by DavidMansolino** [comment]
test me

**Comment by DavidMansolino** [comment]
:+1:

**Pull request merged**